apps/cmp.c

```diff
@@ -1607,15 +1604,13 @@ static int setup_request_ctx(OSSL_CMP_CTX *ctx, ENGINE *engine)
             goto exts_err;
         }
         OSSL_CMP_CTX_set0_reqExtensions(ctx, exts);
-        exts = NULL;
```

Comment:
Is this correct?? set0 means the OSSL_CMP_CTX takes ownership, or not?

Reply:
Ah, I see we do not jump to the err labels from here in the code. I wonder if this does not make the code fragile.

Reply:
Yes, `ctx` takes ownership.
This removal is part of the simplification of the error handling I did along with the fix.
So far, a later `goto oom` had been possible, which I removed, 
so from the above change onward there cannot be an extra free of `exts`.

Reply:
Maybe add a comment that there must be no goto oom afterwards?

Reply:
Yeah, or even better if I re-add the two NULL assignments, just to be on the safe side?

Reply:
I wonder if static analyzers or gcc won't start screaming about value assigned but not used.

Reply:
Good point. So added a warning comment.

Reply:
BTW, I wonder why Coverity did not find/report that potential (but unlikely) double free?

Reply:
Static analysis is never perfect.